### PR TITLE
Fix #5946 - Port-a-Brig teleporting

### LINKED ISSUE #5946: Portabrig "duplicating" people, making them teleport back to it.
[BUG]

**Describe the bug**
I was observing the round "WaveJupiterStar Laboratory 2053" when a vampire got busted. As they broke free of the portabrig, a secofficer tried to put them in the portabrig again. As soon the loading bar to insert them in there finished, they used the flash skill or passed out, I'm not sure, but the portabrig image went from empty to occupied, even though the vampire was still outside. They got stunned again and taken to the shuttle in cuffs. Later, when the round ended, another secoff that stayed behind found the portabrig, still occupied, and maybe unlocked it, or pressed the button to empty it. The vampire, that was being beaten in the shuttle, suddenly was in the station, where the portabrig was.

**To Reproduce**
1. Try to insert someone in the portabrig
2. Have them to pass out or use a vampire flash skill, or both?
3. The animation in the portabrig will change, even though they are still outside
4. Move eject them from the portabrig and they should be teleported to it

**Expected behavior**
When i saw them being inserted in the portabrig, i expected or it being a success, making the portabrig occupied, or it failing, leaving it empty.

**Screenshots**
Screenshot of the vampire on station, with the sec guy that has never met them before.
image

**Additional context**

### PR DESCRIPTION
**Pull request by flappybatpal**
[BUG]

## About the PR 
Fixes #5946

## Why's this needed? 
If for whatever reason, such as recall artifact or admin intervention, a mob was teleported out of the port-a-brig it would still set their loc back to the port-a-brig tile when emptied.

### CODE REVIEW

**Comment by flappybatpal** [comment]
Could change process to always run while occupied and check for occupants, unless there is an event that gets fired by contents changing? I think process might be overkill.

**Comment by Gerhazo** [comment]
Edited to reflect referring to port-a-brig, as it took me a while to understand what the PR was about at first.

**Comment by Tarmunora** [comment]
> 
> 
> Could change process to always run while occupied and check for occupants, unless there is an event that gets triggered by contents changing? I think process might be overkill.

`Exited()` proc

File: code/obj/machinery/porters.dm [hunk 1]
```diff
@@ -385,6 +385,15 @@ var/global/list/portable_machinery = list() // stop looping through world for th
 		src.go_out()
 		return
 
+	Exited(atom/movable/Obj)
+		..()
+		src.occupant = null
```
Comment (Tarmunora) [inline code/obj/machinery/porters.dm:390]: should check that the movable exiting is actually the occupant

**Review by Tarmunora — CHANGES_REQUESTED** [review]

File: code/obj/machinery/porters.dm [hunk 2]
```diff
@@ -387,10 +387,11 @@ var/global/list/portable_machinery = list() // stop looping through world for th
...
-		for (var/obj/item/I in src) //What if you drop something while inside? WHAT THEN HUH?
-			I.set_loc(src.loc)
+		if(Obj == src.occupant)
+			src.occupant = null
+			build_icon()
+			for (var/obj/item/I in src) //What if you drop something while inside? WHAT THEN HUH?
+				I.set_loc(src.loc)
 
 		if (processing)
 			UnsubscribeProcess()
```
Comment (Tarmunora) [inline code/obj/machinery/porters.dm]: indent

Comment (flappybatpal) [reply code/obj/machinery/porters.dm]: Good catch

**Review by Tarmunora — APPROVED** [review]

**Review by UrsulaMejor — APPROVED** [review]

**Pull request merged**